contracts/math/SafeMath.sol

```diff
@@ -7,42 +7,71 @@ pragma solidity ^0.4.18;
  */
 library SafeMath {
 
-  /**
-  * @dev Multiplies two numbers, throws on overflow.
-  */
-  function mul(uint256 a, uint256 b) internal pure returns (uint256) {
-    if (a == 0) {
-      return 0;
+    int256 constant INT256_MIN = int256((uint256(1) << 255));
+
+    int256 constant INT256_MAX = int256(~((uint256(1) << 255)));
```

Comment:
Do we need these constants to check for overflow? Shouldn't the sign be enough?

Reply:
i dont think so, what do you mean with the sign?